src/session/session_api.c

```diff
@@ -488,6 +493,34 @@ err:	API_END_RET_NOTFOUND_MAP(session, ret);
 }
 
 /*
+ * __session_reset --
+ *	WT_SESSION->reset method.
+ */
+static int
+__session_reset(WT_SESSION *wt_session)
+{
+	WT_DECL_RET;
+	WT_SESSION_IMPL *session;
+
+	session = (WT_SESSION_IMPL *)wt_session;
+
+	SESSION_API_CALL_NOCONF(session, reset);
+
+	if (F_ISSET(&session->txn, WT_TXN_RUNNING))
+		WT_ERR_MSG(session, EINVAL, "transaction in progress");
+
+	WT_TRET(__wt_session_reset_cursors(session, 1));
+
+	WT_ASSERT(session, session->ncursors == 0);
+
+	/* Discard scratch buffers, error memory. */
```

Comment:
I don't think this comment adds value, and it's prone to become out of sync with the code.